src/main/kotlin/org/malachite/estella/task/api/TaskController.kt

```diff
@@ -3,107 +3,129 @@ package org.malachite.estella.task.api
 import org.malachite.estella.commons.EStellaHeaders
 import org.malachite.estella.commons.Message
 import org.malachite.estella.commons.OwnResponses
-import org.malachite.estella.commons.PayloadUUID
 import org.malachite.estella.process.domain.TaskDto
 import org.malachite.estella.process.domain.TaskTestCaseDto
-import org.malachite.estella.services.OrganizationService
 import org.malachite.estella.services.TaskService
 import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.http.HttpStatus
 import org.springframework.http.ResponseEntity
 import org.springframework.transaction.annotation.Transactional
 import org.springframework.web.bind.annotation.*
 
 
 @RestController
+@Transactional
 @RequestMapping("/api/tasks")
 class TaskController(
-        @Autowired private val taskService: TaskService,
-        @Autowired private val organizationService: OrganizationService,
+    @Autowired private val taskService: TaskService,
 ) {
 
+    private val bothUUIDErrorMessage = Message("Exactly one of parameters: organizationUuid and taskStageUuid is required")
+
     @Deprecated("Not tested yet - draft implementation, should be tested as part of ES-162")
     @CrossOrigin
-    @Transactional
     @GetMapping
     fun getTasks(
-            @RequestParam("owner", required = false) organizationUuid: String?,
-            @RequestParam("taskStage", required = false) taskStageUuid: String?,
-            @RequestHeader(EStellaHeaders.devPassword) password: String
+        @RequestParam("owner", required = false) organizationUuid: String?,
+        @RequestParam("taskStage", required = false) taskStageUuid: String?,
+        @RequestHeader(EStellaHeaders.devPassword) password: String
     ): ResponseEntity<Any> {
-        if ((organizationUuid == null && taskStageUuid == null) || (organizationUuid != null && taskStageUuid != null))
-            return ResponseEntity.badRequest().body(Message("Exactly one of parameters: organizationUuid and taskStageUuid is required"))
+        if (listOfNotNull(organizationUuid, taskStageUuid).size != 1)
+            return ResponseEntity.badRequest().body(bothUUIDErrorMessage)
         val tasks: List<TaskDto> = organizationUuid
-                ?.let {
-                    taskService.checkDevPassword(it, password)
-                            .getTasksByOrganizationUuid(it)
-                }
-                ?: taskStageUuid
-                        ?.let { taskService.getTasksByTasksStage(it, password) }!!
+            ?.let {
+                taskService.checkDevPassword(it, password)
+                    .getTasksByOrganizationUuid(it)
+            }
+            ?: taskStageUuid?.let { taskService.getTasksByTasksStage(it, password) }!!
```

Comment:
Wydaje mi się że byłoby to wszystko czytelniejsze w when ( włącznie z tym ifem)

Reply:
popieram

Reply:
Tu się nie skupiałem, bo to wjechało w tikecie Tomka

Reply:
W sensie Tomek to i tak przepisywał